### DISCUSSION TITLE
"summary" reporter is not tolerant to system calls of parallel backend

### DISCUSSION THREAD

**Discussion by kkmann** [ropensci/targets#528] [Help]
## Prework

* [x] Read and agree to the [code of conduct](https://ropensci.org/code-of-conduct/) and [contributing guidelines](https://github.com/ropensci/targets/blob/main/CONTRIBUTING.md).
* [x] Confirm that your issue is most likely a genuine bug in the `targets` package itself and not a user error, known limitation, or bug in another package that `targets` depends on. For example, if you get errors running `tar_make_clustermq()`, try isolating the problem in a reproducible example that runs `clustermq` and not `targets`. And for miscellaneous troubleshooting, please post to [discussions](https://github.com/ropensci/targets/discussions) instead of [issues](https://github.com/ropensci/targets/issues).
* [x] If there is [already a relevant issue](https://github.com/ropensci/targets/issues), whether open or closed, comment on the existing thread instead of posting a new issue.
* [ ] ~~Post a [minimal reproducible example](https://www.tidyverse.org/help/) like [this one](https://github.com/ropensci/targets/issues/256#issuecomment-754229683) so the maintainer can troubleshoot the problems you identify. A reproducible example is:~~
    * [ ] ~~**Runnable**: post enough R code and data so any onlooker can create the error on their own computer.~~
    * [ ] ~~**Minimal**: reduce runtime wherever possible and remove complicated details that are irrelevant to the issue at hand.~~
    * [ ] ~~**Readable**: format your code according to the [tidyverse style guide](https://style.tidyverse.org/).~~

## Description

When using `targets::tar_make_future(workers = 1000, reporter = 'timestamp')` in conjunction with the future.batchtools package on a slurm cluster it can happen that the slurm CLI becomes unresponsive for a long enough period so that the reporter throws an error. This then terminates the entire workflow. Execution with other / no reporters are no problem. The summary reporter is particularly useful for long running cluster jobs to get an idea of the work progress.

## Reproducible example

I intentionally skipped the minimal reproducible example since the issue depends on the execution environment and I cannot share the particular workflow openly in any useful way.

## Expected result

I would expect the "summary" reporter to be robust against timeouts to calls to the backend. I.e., a timeout to squeue or whatever is used to query the necessary information should not kill the workflow execution (it is just a reporting error not an execution error!).

## Diagnostic information

* Session info
```
R version 4.0.3 (2020-10-10)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Scientific Linux 7.9 (Nitrogen)

Matrix products: default
BLAS/LAPACK: /usr/lib64/libopenblas-r0.3.3.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices datasets  utils     methods   base     

other attached packages:
[1] future.batchtools_0.10.0 future_1.21.0            targets_0.3.1           

loaded via a namespace (and not attached):
 [1] pillar_1.6.0      compiler_4.0.3    prettyunits_1.1.1 progress_1.2.2   
 [5] tools_4.0.3       digest_0.6.27     lifecycle_1.0.0   tibble_3.1.1     
 [9] checkmate_2.0.0   pkgconfig_2.0.3   rlang_0.4.10      igraph_1.2.6     
[13] cli_2.4.0         parallel_4.0.3    withr_2.4.2       rappdirs_0.3.3   
[17] hms_1.0.0         vctrs_0.3.7       globals_0.14.0    tidyselect_1.1.0 
[21] glue_1.4.2        data.table_1.14.0 listenv_0.8.0     R6_2.5.0         
[25] processx_3.5.1    fansi_0.4.2       parallelly_1.24.0 base64url_1.4    
[29] callr_3.6.0       purrr_0.3.4       magrittr_2.0.1    backports_1.2.1  
[33] codetools_0.2-18  ps_1.6.0          batchtools_0.9.15 ellipsis_0.3.1   
[37] renv_0.11.0       brew_1.0-6        utf8_1.2.1        stringi_1.5.3    
[41] crayon_1.4.1     
```
* installed targets package description
```
> packageDescription("targets")
Package: targets
Title: Dynamic Function-Oriented 'Make'-Like Declarative Workflows
Description: As a pipeline toolkit for Statistics and data science in
        R, the 'targets' package brings together function-oriented
        programming and 'Make'-like declarative workflows.  It analyzes
        the dependency relationships among the tasks of a workflow,
        skips steps that are already up to date, runs the necessary
        computation with optional parallel workers, abstracts files as
        R objects, and provides tangible evidence that the results
        match the underlying code and data. The methodology in this
        package borrows from GNU 'Make' (2015, ISBN:978-9881443519) and
        'drake' (2018, <doi:10.21105/joss.00550>).
Version: 0.3.1
License: MIT + file LICENSE
URL: https://docs.ropensci.org/targets/,
        https://github.com/ropensci/targets
BugReports: https://github.com/ropensci/targets/issues
Authors@R: c( person( given = c("William", "Michael"), family =
        "Landau", role = c("aut", "cre"), email =
        "will.landau@gmail.com", comment = c(ORCID =
        "0000-0003-1878-3253") ), person( given = c("Matthew", "T."),
        family = "Warkentin", role = "ctb" ), person( given =
        "Samantha", family = "Oliver", role = "rev", comment = c(ORCID
        = "0000-0001-5668-1165") ), person( given = "Tristan", family =
        "Mahr", role = "rev", comment = c(ORCID =
        "0000-0002-8890-5116") ), person( family = "Eli Lilly and
        Company", role = "cph" ))
Depends: R (>= 3.5.0)
Imports: callr (>= 3.4.3), cli (>= 2.0.2), codetools (>= 0.2.16),
        data.table (>= 1.12.8), digest (>= 0.6.25), igraph (>= 1.2.5),
        R6 (>= 2.4.1), rlang (>= 0.4.10), stats, tibble (>= 3.0.1),
        tidyselect (>= 1.1.0), utils, vctrs (>= 0.2.4), withr (>=
        2.1.2)
Suggests: arrow (>= 3.0.0), aws.s3 (>= 0.3.21), bs4Dash(>= 0.5.0),
        clustermq (>= 0.8.95.1), curl (>= 4.3), dplyr (>= 1.0.0), fst
        (>= 0.9.2), future (>= 1.19.1), future.callr (>= 0.6.0), gt (>=
        0.2.2), keras (>= 2.2.5.0), knitr (>= 1.30), rmarkdown (>=
        2.4), pingr (>= 2.0.1), pkgload (>= 1.1.0), qs (>= 0.23.2),
        rstudioapi (>= 0.11), shiny (>= 1.5.0), shinycssloaders (>=
        1.0.0), shinyWidgets (>= 0.5.4), testthat (>= 3.0.0), torch (>=
        0.1.0), usethis (>= 1.6.3), visNetwork (>= 2.0.9)
Encoding: UTF-8
Language: en-US
VignetteBuilder: knitr
Config/testthat/edition: 3
RoxygenNote: 7.1.1
NeedsCompilation: no
Packaged: 2021-03-27 16:51:14 UTC; c240390
Author: William Michael Landau [aut, cre]
        (<https://orcid.org/0000-0003-1878-3253>), Matthew T. Warkentin
        [ctb], Samantha Oliver [rev]
        (<https://orcid.org/0000-0001-5668-1165>), Tristan Mahr [rev]
        (<https://orcid.org/0000-0002-8890-5116>), Eli Lilly and
        Company [cph]
Maintainer: William Michael Landau <will.landau@gmail.com>
Repository: CRAN
Date/Publication: 2021-03-28 06:50:02 UTC
Built: R 4.0.3; ; 2021-05-04 11:36:31 UTC; unix
```

## Suggested Fix

* catch exceptions of system calls used for reporting and allow repeated warnings / failures due to timeout without stopping the workflow

**Comment by wlandau** [comment]
Would you show the error message? And would you try the development version of `targets`? Reporters themselves do not get involved in how `targets` or R handles errors, they just print text.

**Reply by wlandau** [reply]
Also, are you able to create a reprex without `tar_make_future()`? Maybe try `tar_make()` on your cluster with thousands of dynamic branches and run it under the same conditions as `tar_make_future()`? The reporter operates the same way whether you use `tar_make_future()` or `tar_make()`. The reporter exists in the main process and only uses information available to the main process. The choice of `tar_make()` vs `tar_make_future()` should make no difference to the reporter except in terms of how fast targets get processed.

**Reply by stuvet** [reply]
Hi @kkmann have you tried again with the latest version ? I strongly suspect your issue was similar to my own from  #568   which @wlandau fixed today with 141859fab4492e0566af257d8256157509ad4e5f before I could even post the reprex.
